src/scene/particle-container/shared/Particle.ts

```diff
@@ -0,0 +1,144 @@
+import { Color } from '../../../color/Color';
+import { Texture } from '../../../rendering/renderers/shared/texture/Texture';
+
+import type { ColorSource } from '../../../color/Color';
+
+/**
+ * Represents a particle with properties for position, scale, rotation, color, and texture.
+ * @property {number} x - The x-coordinate of the particle.
+ * @property {number} y - The y-coordinate of the particle.
+ * @property {number} scaleX - The scale factor in the x-axis.
+ * @property {number} scaleY - The scale factor in the y-axis.
+ * @property {number} anchorX - The x-coordinate of the anchor point.
+ * @property {number} anchorY - The y-coordinate of the anchor point.
+ * @property {number} rotation - The rotation of the particle in radians.
+ * @property {number} color - The color of the particle as a hexadecimal number.
+ * @property {Texture} texture - The texture of the particle.
+ */
+export interface IParticle
+{
+    x: number;
+    y: number;
+    scaleX: number;
+    scaleY: number;
+    anchorX: number;
+    anchorY: number;
+    rotation: number;
+    color: number;
+    texture: Texture;
+}
+
+export type ParticleOptions = Omit<Partial<IParticle>, 'color'> & {
+    texture: Texture
+    tint?: number;
+    alpha?: number;
+};
+
+/**
+ * Represents a single particle within a particle container. This class implements the IParticle interface,
+ * providing properties and methods to manage the particle's position, scale, rotation, color, and texture.
+ *
+ * The reason we use a particle over a sprite is that these are much lighter weight and we can create a lot of them
+ * without taking on the overhead of a full sprite.
+ *
+ * Here is an example of how to create a new particle:
+ *
+ * ```javascript
+ * const particle = new Particle(texture);
+ * particle.x = 100;
+ * particle.y = 100;
+ * particle.scaleX = 0.5;
+ * particle.scaleY = 0.5;
+ * particle.rotation = Math.PI / 2;
+ * particle.color = 0xff0000;
```

Comment:
Preference here for options:
```suggestion
 * const particle = new Particle({
 *   texture,
 *   x: 100,
 *   y: 100,
 *   scaleX: 0.5,
 *   scaleY: 0.5,
 *   rotation: Math.PI / 2,
 *   color: 0xff0000,
 * });
```